Common/Securities/CashAccountMarginModel.cs

```diff
@@ -0,0 +1,242 @@
+        public decimal GetInitialMarginRequiredForOrder(Security security, Order order)
+        {
+            var fees = security.FeeModel.GetOrderFee(security, order);
+            var marginRequiredInAccountCurrency = order.GetValue(security) + fees;
+
+            switch (order.Direction)
+            {
+                case OrderDirection.Hold:
+                case OrderDirection.Buy:
+                    // convert the order value to the quote currency
+                    return marginRequiredInAccountCurrency / security.QuoteCurrency.ConversionRate;
+
+                case OrderDirection.Sell:
+                    // remaining margin in units of the base currency ( can't sell what we don't have )
+                    var baseCurrency = security as IBaseCurrencySymbol;
+                    if (baseCurrency != null)
+                    {
+                        // we have this much 'cash' that can be sold
+                        var feesInBaseCurrency = _cashBook.Convert(fees, CashBook.AccountCurrency, baseCurrency.BaseCurrencySymbol);
+                        return order.AbsoluteQuantity + feesInBaseCurrency;
+                    }
+
+                    // assuming these are 'units' of stock priced in the account currency
+                    return marginRequiredInAccountCurrency;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Gets the margin cash available for a trade in units of the currency required.
+        /// When increasing a position, this is the quote currency, when decreasing a position this is the base currency
+        /// </summary>
+        /// <param name="portfolio">The algorithm's portfolio</param>
+        /// <param name="security">The security to be traded</param>
+        /// <param name="direction">The direction of the trade</param>
+        /// <returns>The margin available for the trade</returns>
+        public decimal GetMarginRemaining(SecurityPortfolioManager portfolio, Security security, OrderDirection direction)
+        {
+            var baseCurrency = security as IBaseCurrencySymbol;
+            var targetCurrency = string.Empty;
+            var openOrdersTotalValue = 0m;
+            var openOrdersTotalQuantity = 0m;
+            var symbolDirectionPairs = new Dictionary<Symbol, OrderDirection>();
+
+            if (baseCurrency != null)
+            {
+                // find the target currency for the requested direction and the securities potentially involved
+                targetCurrency = direction == OrderDirection.Buy
+                    ? security.QuoteCurrency.Symbol
+                    : baseCurrency.BaseCurrencySymbol;
+
+                foreach (var portfolioSecurity in portfolio.Securities.Values)
+                {
+                    var basePortfolioSecurity = portfolioSecurity as IBaseCurrencySymbol;
+                    if (basePortfolioSecurity == null) continue;
```

Comment:
I think we need to remove this check. Specifically in the case of crypto this will always be true, but in the general case it doesn't need to be. Consider an equity portfolio with assets priced in USD and other assets priced in EUR. Since they're equities, they won't have a base currency symbol, but we'll still need to potentially check them.

Maybe this condition should instead be
```csharp
if (basePortfolioSecurity == null && direction == OrderDirection.Sell) { /*...*/ }
```

The thinking here is that when we're selling we care about the base currency, so if we're selling and there is no base currency, it should be safe to skip.